modules/programs/poetry.nix

```diff
@@ -0,0 +1,48 @@
+{ pkgs, config, lib, ... }:
+let
+  tomlFormat = pkgs.formats.toml { };
+  configDir = if pkgs.stdenv.isDarwin then
+    "Library/Application Support"
+  else
+    config.xdg.configHome;
+  cfg = config.programs.poetry;
+in {
+  meta.maintainers = with lib.maintainers; [ mirkolenz ];
+  options.programs.poetry = with lib; {
+    enable = mkEnableOption "poetry";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.poetry;
+      defaultText = literalExpression "pkgs.poetry";
+      description = "The poetry package to use (e.g., with custom plugins).";
+      example = literalExpression
+        "pkgs.poetry.withPlugins (ps: with ps; [ poetry-plugin-up ])";
+    };
```

Comment:
Please use `mkPackageOption`.

Reply:
Thanks for the suggestion, I was not aware this function exists. I updated the option accordingly.